File: Backend/list_models.py
import urllib.request
import json

api_key = "AIzaSyAo1DJBV5HGlXEomUbfVASbD_JsdSzrPm8"
url = f"https://generativelanguage.googleapis.com/v1beta/models?key={api_key}"

try:
    with urllib.request.urlopen(url) as response:
        if response.status == 200:
            data = json.loads(response.read().decode())
            models = data.get('models', [])
            with open('D:/Desktop/NewWebsite/New Web/Backend/models_list.txt', 'w', encoding='utf-8') as f:
                for m in models:
                    name = m['name']
                    if 'flash' in name.lower() or 'pro' in name.lower():
                         f.write(f"{name}\n")
            print("Wrote models to models_list.txt")
        else:
            print(f"Error: {response.status}")
except Exception as e:
    print(f"Exception: {e}")

File: Backend/list_models.js
const fs = require('fs');

try {
    const data = fs.readFileSync('D:/Desktop/NewWebsite/New Web/Backend/gemini_models.json', 'utf8');
    const json = JSON.parse(data);
    if (json.models) {
        console.log('Available Models:');
        json.models.forEach(m => console.log(m.name));
    } else {
        console.log('No models found in JSON.');
    }
} catch (err) {
    // If utf8 fails, try utf16le? Or maybe log the error
    console.error('Error reading file:', err.message);
    // fallback: try simple regex on raw string if JSON parse fails due to encoding
}
